src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -39,6 +39,14 @@ import { relativeToAbsolute } from '../../nav_links/to_nav_link';
 export const isModifiedOrPrevented = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
   event.metaKey || event.altKey || event.ctrlKey || event.shiftKey || event.defaultPrevented;
 
+// TODO: replace hard-coded values with a registration function, so that apps can control active nav links similar to breadcrumbs
+const aliasedApps: { [key: string]: string[] } = {
+  discover: ['data-explorer'],
+};
+
+export const isActiveNavLink = (appId: string | undefined, linkId: string) =>
+  appId === linkId || aliasedApps[linkId]?.includes(appId || '');
```

Comment:
```suggestion
export const isActiveNavLink = (appId: string | undefined, linkId: string): boolean =>
  !!(appId === linkId || aliasedApps[linkId]?.includes(appId || ''));
```
`appId === linkId` used to always return a boolean value, while isActiveNavLink is not. Recommend to keep the behavior as-is.

Reply:
@SuZhou-Joe  That sounds reasonable. Thanks!